### ISSUE TITLE
Run expectations even when fetch has been resolved

### ISSUE THREAD

**Issue by jcassee** [vlucas/frisby#452]
I am trying to use frisby with cucumber to test an API. I am running into trouble separating steps.

Take for example this steps file:
```javascript
const { When, Then, After } = require('cucumber');
const frisby = require('frisby');

When('you GET the resource', function() {
  this.response = frisby.get('/');
})

Then('the response status code is {int}', function(status) {
  return this.response.expect('status', status);
})
```

I have a feature that look like this:
```gherkin
  Scenario: Requesting the resource
    When you GET the resource
    Then the response status code is 200
    And the response status code is 201
```

However, because the first `Then` step resolves the fetch and the expectations run only when the fetch is resolved, the second test for the status code does not run and the scenario succeeds. It is possible to remove the `return` keyword from the `Then` step and add this:

```javascript
After(function (scenario) {
  return this.response
})
```

But in that case cucumber will only fail at the `After` step, and not at the `Then` that really failed, which is confusing.

Would it be possible to run the expectations when calling `response.expect` in case the fetch promise was already resolved?

**Comment by H1Gdev** [comment]
I will consider this issue.

**Comment by vlucas** [comment]
Published `v2.0.15` with this fix
